zeroeq/fbevent.cpp

```diff
@@ -0,0 +1,84 @@
+
+/* Copyright (c) 2014-2015, Human Brain Project
+ *                          Daniel Nachbaur <daniel.nachbaur@epfl.ch>
+ */
+
+#include "fbevent.h"
+#include "detail/fbevent.h"
+
+namespace zeroeq
+{
+
+FBEvent::FBEvent( const uint128_t& type, const EventFunc& func )
+    : _impl( new detail::FBEvent( type, func ))
+{
+    if( func )
+        registerDeserializedCallback( [this](){ _impl->func( *this ); } );
+}
+
+FBEvent::~FBEvent()
+{
+}
+
+std::string FBEvent::getTypeName() const
+{
+    return _impl->type.getString();
```

Comment:
This is used by the serializable. TypeName is converted to typeidentifier in servus::serializable

uint128_t Serializable::getTypeIdentifier() const
{
    return make_uint128( getTypeName( ));
}